library/src/main/java/com/schibsted/spain/barista/interaction/PermissionGranter.kt

```diff
@@ -36,15 +35,43 @@ object PermissionGranter {
         sleepThread(PERMISSIONS_DIALOG_DELAY.toLong())
         val device = UiDevice.getInstance(getInstrumentation())
 
-        val resourceId = if (permissionNeeded == ACCESS_FINE_LOCATION || permissionNeeded == ACCESS_COARSE_LOCATION) {
-            PERMISSIONS_DIALOG_ALLOW_FOREGROUND_ID
-        } else {
-            PERMISSIONS_DIALOG_ALLOW_ID
+        val regex = PERMISSION_DIALOG_ALLOW_FOREGROUND_IDS.joinToString(
+          prefix = "^(",
+          separator = "|",
+          postfix = ")$"
+        ) { it }
+        val allowPermissions = device.findObject(UiSelector()
+            .clickable(true)
+            .checkable(false)
+            .resourceIdMatches(regex)
+        )
+        if (allowPermissions.exists()) {
+          allowPermissions.click()
         }
+      }
+    } catch (e: UiObjectNotFoundException) {
+      Log.e("Barista", "There is no permissions dialog to interact with", e)
+    }
+  }
+
+  @JvmStatic
+  fun allowPermissionOneTime(permissionNeeded: String) {
```

Comment:
If we're adding this new method it should have a test to cover it at least 😅

Reply:
`ignores_already_granted_permission` uses it

Reply:
Hmm and I guess we can't "actually" test the behaviour of that button. I mean, check that next time the permission is not granted. Because we can't reproduce that from one test. 🤔 😞 
Well, we will have to trust Android on this one 😄